plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/cache/IcebergCacheKeyProvider.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.cache;
+
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.cache.CacheKeyProvider;
+
+import java.io.IOException;
+import java.util.Optional;
+
+public class IcebergCacheKeyProvider
+        implements CacheKeyProvider
+{
+    @Override
+    public Optional<String> getCacheKey(TrinoInputFile delegate)
+            throws IOException
+    {
+        String path = delegate.location().path();
+        if (path.endsWith(".trinoSchema") || path.contains("/.trinoPermissions/")) {
```

Comment:
Unrelated to this PR:
Do we have a way to know which existing or future special files or paths might be present in a table/accessed by the connector? The Delta Lake connector is unfortunately not aware of these files at the moment and will happily cache them, which might have bad consequences.

Instead of typically caching everything by default, it would be better if we cached nothing by default, and then had a way to specify which file types should be cached. This might be involved to do though. An alternative could be to make caching on the coordinator very restrictive.

Reply:
These files are generated only by FileHiveMetastore which is meant only for tests. So this shouldn't be used in the main source code, it should handled in tests similar to TestIcebergFileOperations

Reply:
this is removed now.